src/sentry/snuba/metrics/query_builder.py

```diff
@@ -406,24 +406,32 @@ def is_tag_key_allowed(tag_key: str, allowed_tag_keys: Optional[Dict[str, str]])
     return allowed_tag_keys.get(f"tags[{tag_key}]") is not None
 
 
-def parse_query(query_string: str, projects: Sequence[Project]) -> Sequence[Condition]:
-    """Parse given filter query into a list of snuba conditions"""
+def parse_conditions(
+    query_string: str, projects: Sequence[Project], query_params: Mapping[str, Any]
+) -> Sequence[Condition]:
+    """Parse given filter query and query params into a list of snuba conditions"""
     # HACK: Parse a sessions query, validate / transform afterwards.
     # We will want to write our own grammar + interpreter for this later.
     try:
         query_builder = ReleaseHealthQueryBuilder(
             Dataset.Sessions,
             params={
+                **query_params,
                 "project_id": [project.id for project in projects],
                 "organization_id": org_id_from_projects(projects) if projects else None,
             },
             config=QueryBuilderConfig(use_aggregate_conditions=True),
         )
         where, _ = query_builder.resolve_conditions(query_string)
+        param_conditions = list(
+            filter(
+                lambda condition: condition.lhs.name != "project_id", query_builder.resolve_params()
```

Comment:
With this condition are you trying to remove the possible duplicate condition for `project_id`?

Reply:
In addition, `resolve_params` returns a `Union` type made of `Condition` and `BooleanCondition` thus you should make a function that traverses the types and removes the project. If a user uses an `OR` or `AND` the filtering closure will fail.

Reply:
Something like this:

```python
def _strip_project_id(cls, condition):
    if isinstance(condition, BooleanCondition):
        new_boolean_condition = BooleanCondition()
        for nested_condition in condition.conditions:
            stripped_condition = _strip_project_id(nested_condition)
            # Only if we have a condition object we append, since if its None it means that we directly had a
            # project.id.
            # You can also optimize this code by returning a normal condition in case the number of conditions is = 1 after stripping.
            if stripped_condition is not None:
                new_boolean_condition.conditions.append(stripped_condition)

        return new_boolean_condition
    elif isinstance(condition, Condition):
        if isinstance(condition.lhs, Column):
            if condition.lhs.name == "project_id":
                return None

            return condition
```

Reply:
This is overkill if you know for a fact that `project_id` will be top-level only but I assume you can also have a query like `project:1 AND release:foo` which will have to be transformed to `release:foo`.

Reply:
Left additional comments, thanks for fixing the problems!